code/gateway/dataaccessOverwrite.q

```diff
@@ -0,0 +1,71 @@
+\d .dataaccess
+
+// Dynamic routing finds all processes with relevant data
+attributesrouting:{[options;procdict]
+    // Get the tablename and timespan
+    timespan:`long$options[`starttime`endtime];
+    //adjust rdb partition range to account for period end time)
+    if[`rdb in key procdict;procdict[`rdb]:(first[procdict `rdb];-1+`long$01D00 + last procdict `rdb)];
```

Comment:
The int partitions are defined as of the period start time so if we attempt to search for data in the partition the api wont recognise it as being within the partition (works with normal set up and it compares days i.e 2024.03.11 in 2024.03.11 = 1b). Need to then establish a valid range for the current rdb partition which is why ive changed the max value the partition dict to be just beforew days end. Say current rdb partition is 10:00:00 then this will return `rdb| 10:00:00 23:59:59 and then api will recognise that data after 10 is in the rdb

Reply:
chose days end to cover all possible periods up to 24hrs